File: WeekendWarriors/Questionnaires/Questionnaires/QLS/AST/Widgets/Spinbox.cs
using Questionnaires.Renderer.Widgets;

namespace Questionnaires.QLS.AST.Widgets
{
    class Spinbox : Widget
    {
        public override QuestionWidget CreateWidget(Types.IntegerType type)
        {
            return new IntegerPickerWidget();
        }

        public override QuestionWidget CreateWidget(Types.MoneyType type)
        {
            return new DecimalPickerWidget();
        }
    }
}

File: AlternativeQL/src/org/qls/ast/widget/Spinbox.java
package org.qls.ast.widget;

import org.ql.ast.type.*;

public class Spinbox extends Widget {

    public boolean isCompatibleWith(FloatType type) {
        return true;
    }

    public boolean isCompatibleWith(IntegerType type) {
        return true;
    }

    public boolean isCompatibleWith(MoneyType type) {
        return true;
    }

    @Override
    public <T, C> T accept(WidgetVisitor<T, C> visitor, C context) {
        return visitor.visitSpinbox(this, context);
    }
}
